File: EPRO.Library.v3.5/EPRO.Library.v3.5/Objects/EShort.cs
using System;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace ELibrary.Standard.Objects
{
    public sealed class EShort
    {

        /// <summary>
        /// Converts An Object to long
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static short valueOf(string obj)
        {
            try
            {
                if (Information.IsNothing(obj))
                    return 0;
                if (string.IsNullOrEmpty(obj))
                    return 0;
                if (string.IsNullOrEmpty(obj.Trim()))
                    return 0;
                return Conversions.ToShort(obj);
            }
            catch (Exception ex)
            {
                return 0;
            }
        }


        /// <summary>
        /// Converts An Object to short
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static short valueOf(DBNull obj)
        {
            return 0;
        }

        /// <summary>
        /// Converts An Object to Short
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static short valueOf(double obj)
        {
            try
            {
                return (short)Math.Round(obj);
            }
            catch (Exception ex)
            {
                return 0;
            }
        }


        /// <summary>
        /// Converts An Object to Short
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static short valueOf(decimal obj)
        {
            try
            {
                return (short)Math.Round(obj);
            }
            catch (Exception ex)
            {
                return 0;
            }
        }

        /// <summary>
        /// Converts An Object to Short
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static short valueOf(int obj)
        {
            try
            {
                return (short)obj;
            }
            catch (Exception ex)
            {
                return 0;
            }
        }


        /// <summary>
        /// Converts An Object to Short
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static short valueOf(long obj)
        {
            try
            {
                return (short)obj;
            }
            catch (Exception ex)
            {
                return 0;
            }
        }



        /// <summary>
        /// Converts boolean to short
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static short valueOf(bool obj)
        {
            try
            {
                return (short)Math.Round(Math.Abs(Conversion.Val(obj)));
            }
            catch (Exception ex)
            {
                return 0;
            }
        }

        /// <summary>
        /// Converts objects to short
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static short valueOf(object obj)
        {
            if (obj is null)
                return 0;
            if (obj is string)
                return valueOf(Conversions.ToString(obj));
            if (obj is DBNull)
                return valueOf((DBNull)obj);
            if (obj is double)
                return valueOf(Conversions.ToDouble(obj));
            if (obj is long)
                return valueOf(Conversions.ToLong(obj));
            if (obj is int)
                return valueOf(Conversions.ToInteger(obj));
            if (obj is short)
                return Conversions.ToShort(obj);
            if (obj is bool)
                return valueOf(Conversions.ToBoolean(obj));
            return 0;    // REM Cant convert this
        }
    }
}

File: ELibrary.Standard.VB/Objects/EShort.vb


Imports ELibrary.Standard.VB.Modules

Namespace Objects

    Public NotInheritable Class EShort

        ''' <summary>
        ''' Converts An Object to long
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As String) As Int16
            Try

                If basExtensions.IsNothing(obj) Then Return 0
                If obj = "" Then Return 0
                If obj.Trim = "" Then Return 0

                Return CShort(obj)

            Catch ex As Exception
                Return 0

            End Try

        End Function


        ''' <summary>
        ''' Converts An Object to short
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As System.DBNull) As Int16
            Return 0
        End Function

        ''' <summary>
        ''' Converts An Object to Short
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Double) As Int16
            Try

                Return CShort(obj)

            Catch ex As Exception
                Return 0

            End Try

        End Function


        ''' <summary>
        ''' Converts An Object to Short
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Decimal) As Int16
            Try

                Return CShort(obj)

            Catch ex As Exception
                Return 0

            End Try



        End Function

        ''' <summary>
        ''' Converts An Object to Short
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Int32) As Int16
            Try

                Return CShort(obj)

            Catch ex As Exception
                Return 0

            End Try



        End Function


        ''' <summary>
        ''' Converts An Object to Short
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Long) As Int16
            Try

                Return CShort(obj)

            Catch ex As Exception
                Return 0

            End Try



        End Function



        ''' <summary>
        ''' Converts boolean to short
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Boolean) As Int16
            Try

                Return CShort(Math.Abs(CInt(obj)))

            Catch ex As Exception
                Return 0

            End Try

        End Function

        ''' <summary>
        ''' Converts objects to short
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Object) As Short

            If obj Is Nothing Then Return 0

            If TypeOf obj Is String Then Return valueOf(CType(obj, String))
            If TypeOf obj Is DBNull Then Return valueOf(CType(obj, DBNull))
            If TypeOf obj Is Double Then Return valueOf(CType(obj, Double))
            If TypeOf obj Is Long Then Return valueOf(CType(obj, Long))
            If TypeOf obj Is Int32 Then Return valueOf(CType(obj, Int32))
            If TypeOf obj Is Int16 Then Return CType(obj, Int16)
            If TypeOf obj Is Boolean Then Return valueOf(CType(obj, Boolean))

            Return 0    REM Cant convert this

        End Function


    End Class

End Namespace